python2/koans/about_string_manipulation.py

```diff
@@ -10,38 +10,38 @@ def test_use_format_to_interpolate_variables(self):
         value1 = 'one'
         value2 = 2
         string = "The values are {0} and {1}".format(value1, value2)
-        self.assertEqual(__, string)
+        self.assertEqual("The values are one and 2", string)
 
     def test_formatted_values_can_be_shown_in_any_order_or_be_repeated(self):
         value1 = 'doh'
         value2 = 'DOH'
         string = "The values are {1}, {0}, {0} and {1}!".format(value1, value2)
-        self.assertEqual(__, string)
+        self.assertEqual("The values are DOH, doh, doh and DOH!", string)
 
     def test_any_python_expression_may_be_interpolated(self):
         import math  # import a standard python module with math functions
 
         decimal_places = 4
         string = "The square root of 5 is {0:.{1}f}".format(math.sqrt(5), \
             decimal_places)
-        self.assertEqual(__, string)
+        self.assertEqual("The square root of 5 is 2.2361", string)
```

Comment:
> 27 string = "The square root of 5 is {0:.{1}f}".format(math.sqrt(5), \decimal_places)
> I'm not sure about what the {0:.{1}f exactly does

See [this](https://docs.python.org/3.4/library/string.html#formatstrings) and [this](https://docs.python.org/3.4/library/string.html#formatspec).

In short `{0:.{1}f}` is saying: get the argument `{0}`, but add formatting to it, interpreting as a floating point value (the `f`) and such that the precision (number of digits after the decimal point) is whatever the value of the `{1}` argument says (in this case, 4).

I hope that helps. Play with it a little, it's very similar to C-language string formatting.

Reply:
Given:
```
days = dict()
days["Mercury"] = 87.96
days["Venus"] = 224.68
days["Earth"] = 365.26
days["Mars"] = 686.98
```
Printing this with:
```
for key,value in days.items():
     print("{0} {1}".format(key, value))
```
Looks ugly for a user interface (try). Give me a code snippet that prints this in tidy columns, using 10 characters for the planet name field width, and 20 for the period revolution days, with 3 for the precision. It should look like this:
```
Mercury        87.960
Earth         365.260
Venus         224.680
Mars          686.980
```

Reply:
I think is ordering by the key with a higher value, but I'm not sure.

Reply:
> I think is ordering by the key with a higher value, but I'm not sure.

Think again https://github.com/stevendes/python_koans/pull/6/files#diff-7ff6c7eb24ff198b3e2fab6c93748dd2R36

Reply:
What about the code I asked above?

Reply:
So it's because the dictionary doesn't really have an order?